### DISCUSSION TITLE
Ability to constrain a UUID to a specific version

### DISCUSSION THREAD

**Discussion by jesselang** [cue-lang/cue#2580] [Q&A]
While there is a constraint for a valid UUID, there isn't a constraint for a particular version.

```
import (
"uuid"
)

id: uuid.Valid
id: "1c7b233b-4806-41a7-bff2-8feebef4870f"
```

**Describe the solution you'd like**

Something like:

```
import (
"uuid"
)

id: uuid.Valid & uuid.Version(4)
id: "1c7b233b-4806-41a7-bff2-8feebef4870f"
```

**Describe alternatives you've considered**

There is an operation in the uuid package that returns the version, but I couldn't find a good way to use this as a constraint.

**Additional context**

**Comment by mvdan** [comment]
How about: https://tip.cuelang.org/play/?id=0Yg5UD02ouh#cue@export@cue

Replace the `== 4` with `== 3` for example, and you'll see the failure.

With the `must` builtin per #575, you could rewrite `_isV4: true & uuid.Version(id) == 4` into `_isV4: must(uuid.Version(id) == 4, "id must be an UUID v4")`, for example. A bit nicer for sure, although what you can do today still gets the job done.

**Reply by jesselang** [reply]
Yeah, using another boolean field will work, but the error message isn't going to help a user at all. It seems like the `must` builtin would provide the ability to specify a meaningful error message. Glad to know there is something in the works to wrap these functions into constraints. Thanks!

**Comment by mvdan** [comment]
I'm going to convert this into a discussion, because I feel like it's a valid question that others might find useful in the future, and I don't want to close it as a duplicate issue of the `must` built-in :)
